numpy/core/meson.build

```diff
@@ -594,9 +585,216 @@ np_core_dep = declare_dependency(
     '.',
     'include',
     'src/common',
-  ]
+  ],
+  dependencies: CPU_DEPENDENCY
 )
 
+# Build dispatch-able sources
+# -----------------------------
+CPU_CONF_DISPATCH_SOURCES = {
+  '_simd': {
+    'dependencies': declare_dependency(
```

Comment:
Using `declare_dependency` like this I'm not 100% sure about. Why can't `src_file.process('src/_simd/_simd_inc.h.src'),` be a regular entry in `sources` of the extension module, and same for `include_directories: 'src/_simd/',` etc.?  The code comment says "Build dispatch-able sources", but it doesn't actually get built until we hit `py.extension_module('_simd'`, right?

I suspect it's because different source files have different `targets` entries and get combined into a single `extension_module`, is that right?

Reply:
> The code comment says "Build dispatch-able sources", but it doesn't actually get built until we hit py.extension_module('_simd', right?

no, the second script https://github.com/numpy/numpy/blob/8b514f1edf28b01b800b87d4fff7c0c7b34658ea/numpy/core/meson_cpu/meson.build will parse `CPU_CONF_DISPATCH_SOURCES`, and create

 - `static_library` for each target
 - Generate config files `*.dispatch.h` that required by the runtime call

https://github.com/numpy/numpy/blob/8b514f1edf28b01b800b87d4fff7c0c7b34658ea/numpy/core/meson.build#L791-L797

Then through variable `CPU_GENERATED_DEPS`, we can add these static libraries to the module target. 

https://github.com/numpy/numpy/blob/8b514f1edf28b01b800b87d4fff7c0c7b34658ea/numpy/core/meson.build#L1056-L1063

> I suspect it's because different source files have different targets entries and get combined into a single extension_module, is that right?

yes, that's correct.